cylc/flow/platforms.py

```diff
@@ -63,10 +70,21 @@ def get_platform(task_conf=None, task_id='unknown task', warn_only=False):
         output = platform_from_name(task_conf)
 
     elif 'platform' in task_conf and task_conf['platform']:
+        if REC_COMMAND.match(task_conf['platform']) and warn_only:
+            # In warning mode this function might have been passed an
+            # un-expanded platform string - warn that they won't deal with
+            # with this until job submit.
+            LOG.warning(
+                f'Cannot attempt check validity of [runtime][{task_id}]'
+                f'platform = {task_conf["platform"]}: Platform lookup may fail'
+                ' on job-submit.'
+            )
```

Comment:
```suggestion
```

Reply:
Why?

Reply:
If I'm on the right track - this message would appear for every task which uses the subshell syntax. Saying that platform lookup may fail on job submit is kinda just documenting what this subshell approach does, but once for each task and in a way that few people will likely understand.